external/wakaama/examples/shared/tinydtls/dtls.c

```diff
@@ -3647,21 +3647,24 @@ dtls_handle_message(dtls_context_t *ctx,
     switch (msg[0]) {
 
     case DTLS_CT_CHANGE_CIPHER_SPEC:
-      if (peer) {
-        dtls_stop_retransmission(ctx, peer);
-      }
-      err = handle_ccs(ctx, peer, msg, data, data_length);
-      if (err < 0) {
-	dtls_warn("error while handling ChangeCipherSpec message\n");
-	dtls_alert_send_from_err(ctx, peer, session, err);
-
-	/* invalidate peer */
-	dtls_destroy_peer(ctx, peer, 1);
-	peer = NULL;
-
-	return err;
-      }
-      break;
+		if (peer) {
+			dtls_stop_retransmission(ctx, peer);
+		}
+		if (!peer || !peer->handshake_params) {
```

Comment:
In my opinion, we should do these Valid Pointer checks inside the function which does Dereference.     Or else at how many places, developer has to keep checking the validity of pointer ? 

For ex.   
handle_ccs( )
{
          if( !peer || !peer->handshake_params)
          {
                      return;
          }
}

Another ex:
dtls_stop_retransmission()
{
        if (!peer)
         {
                     return;
          }
}

This way we can keep more reliable code.  What's your opinion ?

Reply:
Basically I agree with you. but in some cases input parameter doesn't need to be checked if all of them have a right value. in that case checking validation inside callee function could have overhead. 
So I think the commit can be acceptable depends on a situation.